addons/mail/static/src/components/dialog/dialog.js

```diff
@@ -65,16 +61,7 @@ export class Dialog extends Component {
      * @param {MouseEvent} ev
      */
     _onClickGlobal(ev) {
-        if (this._componentRef.el && this._componentRef.el.contains(ev.target)) {
```

Comment:
need to keep this line no? that's probably why the guard were needed (and they just hide the problem)

Reply:
Indeed, this was missing. I reintroduced this with logic in models (`.contains()` for the content of dialog).

In practice there were some crashes with attachment viewer on close. I think it's appropriate to add them, because before changes it was executing code with `this._isDragging` and silently worked, but it makes hard crash with `this.attachmentViewer.isDragging` due to non existing `this.attachmentViewer` on deletion.